coda_v2_python_client/firebase_client_wrapper.py

```diff
@@ -36,3 +36,61 @@ def init_client(cls, crypto_token_path, app_name="CodaV2Client"):
         cred = credentials.Certificate(crypto_token_path)
         app = firebase_admin.initialize_app(cred, name=app_name)
         return cls(firestore.client(app))
+
+    def get_dataset_ids():
+        """
+        Gets unique ids of datasets having included only one id for segmented dataset ids.
+
+        :return: Unique ids of datasets having included only one id for segmented dataset ids.
+        :rtype: set of str
+        """
+        segment_ids = get_segment_ids()
+        assert len(segment_ids) == len(set(segment_ids)), "Segment ids not unique"
+
+        dataset_ids = set(segment_ids)
+        for dataset_id in get_segmented_dataset_ids():
+            segment_count = get_segment_count(dataset_id)
+            if segment_count is not None and segment_count > 1:
+                for segment_index in range(2, segment_count + 1):
+                    dataset_ids.remove(id_for_segment(dataset_id, segment_index))
+
+        return dataset_ids
+
+    def get_segment_ids():
+        """
+        Gets ids of all datasets including segmented dataset ids.
+
+        :return: Ids of all datasets including segmented dataset ids.
+        :rtype: list of str
+        """
+        ids = []
+        for segment in self._client.collection(u'datasets').get():
```

Comment:
```suggestion
        for segment in self._client.collection("datasets").get():
```
we only need to support py3